### PR TITLE
Add artifacts and logic related to CORS migration

### PR DESCRIPTION
**Pull request by Buddhimah**

### CODE REVIEW

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        } else {
+            log.info("Configuration Management tables already exist in the database. Hence skipping.");
+        }
+
+    }
+
+    /**
+     * Checks whether Configuration Management database tables are created.
+     *
+     * @return <code>true</core> if checkSQL is success, else <code>false</code>.
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:42]:
```suggestion
     * @return <code>true</code> if checkSQL is success, else <code>false</code>.
```

**Comment by Buddhimah** [comment]
Also address issue: https://github.com/wso2/product-is/issues/7995

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 1 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: done

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package org.wso2.carbon.is.migration.service.v5110.migrator;
```
Comment (pamodaaw) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:1]: License Header missing

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    /**
+     * Checks whether Configuration Management database tables are created.
+     *
+     * @return <code>true</code> if Configuration Management database tables exist, else <code>false</code>.
+     */
+    private boolean isConfigurationManagementDatabaseStructureCreated() throws MigrationClientException {
+
+        try (Connection conn = getDataSource().getConnection();
```
Comment (pamodaaw) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:46]: Don't we need to commit the transaction?

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+             Statement statement = conn.createStatement()) {
+            log.info("Running a query to test the database configuration management tables existence. ");
+            // check whether the tables are already created with a query.
+            try (ResultSet ignored = statement.executeQuery(DB_CHECK_SQL)) {
+                log.info("Configuration Management tables already exists.");
```
Comment (pamodaaw) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:51]: tables already **exist**

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * Checks whether Configuration Management database tables are created.
+     *
+     * @return <code>true</code> if Configuration Management database tables exist, else <code>false</code>.
+     */
+    private boolean isConfigurationManagementDatabaseStructureCreated() throws MigrationClientException {
+
+        try {
+            Connection conn = getDataSource().getConnection();
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:61]: Connection, Statement and resultset closing should be done. rollback on exception should be done.

Please use try-with-resources

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    @Override
+    public void migrate() throws MigrationClientException {
+
+        if (!isConfigurationManagementDatabaseStructureCreated()) {
+            log.info("Configuration Management tables does not exist in the database. Hence creating. ");
+            super.migrate();
+        } else {
+            log.info("Configuration Management tables already exist in the database. Hence skipping.");
+        }
+
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:50]:
```suggestion
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    public void dryRun() throws MigrationClientException {
+
+        log.info("Dry run capability not implemented in {} migrator.", this.getClass().getName());
+    }
+
+    @Override
+    public void migrate() throws MigrationClientException {
+
+        if (!isConfigurationManagementDatabaseStructureCreated()) {
+            log.info("Configuration Management tables does not exist in the database. Hence creating. ");
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:45]:
```suggestion
            log.info("Configuration Management tables does not exist in the database. Hence creating.");
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 2 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java:16]: Done

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 3 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: Done

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 4 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: DOne

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 5 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: Used try with resources to close the connection and statement. Also handled the rollback

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 6 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: Done

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java (hunk 7 above)
Comment (Buddhimah) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v5110/migrator/ConfigurationManagementSchemaMigrator.java]: Done

**Review by omindu — APPROVED** [review]

**Comment by Buddhimah** [comment]
DB Scripts Tested Versions

Oracle 12c
postgres:11.9
mssql 2019
db2:11.5.0.0
mysql:5.7

**Pull request merged**